### DISCUSSION TITLE
Prevent View Transitions when hasUAVisualTransition is set to true

### DISCUSSION THREAD

**Discussion by stefanvonderkrone** [remix-run/react-router#12436] [Proposals]
Hi dear react router team,
I'd like to propose that the current implementation of view transitions should be adjusted to accommodate for the `hasUAVisualTransition` property of the PopStateEvent. Currently, when I swipe back to a previous history state (that had a view transition) the transition will start over and the UI looks weird. There was a manual transition by the user so there is no need for the View Transition at all.
I believe the POP Action is the culprit.

In the video below from my iPhone you see what I mean.

https://github.com/user-attachments/assets/e6bb6d9c-19c9-4e9e-8a06-034dd6fec6fa

Thank you very much.

**Comment by joeuy-dev** [comment]
This would be great!

**Comment by benfrain** [comment]
Yes, please, came here looking for this exact thing. A user swiping back in Safari effectively gets a 'double animation'.

**Comment by bopazyn** [comment]
Support for `hasUAVisualTransition` support would be awesome!

Found good explanation in google docs: 
https://developer.chrome.com/docs/web-platform/view-transitions/same-document#integration_with_the_navigation_api_and_other_frameworks
